nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java

```diff
@@ -170,8 +178,49 @@ public void shouldFailWhenRetrievingDocumentBlobWithoutColdStorageContent() {
         }
     }
 
-    protected DocumentModel createDocument(boolean addBlobContent) {
-        DocumentModel documentModel = session.createDocumentModel("/", "anyFile", "File");
+    @Test
+    public void shouldCheckAvailabilityOfColdStorageContent() {
+        List<String> documents = Arrays.asList( //
+                moveAndRetrieveColdStorageContent(DEFAULT_DOC_NAME).getId(),
+                moveAndRetrieveColdStorageContent("anyFile2").getId(), //
+                moveAndRetrieveColdStorageContent("anyFile3").getId());
+        transactionalFeature.nextTransaction();
+        try (CapturingEventListener listener = new CapturingEventListener(
+                ColdStorageHelper.COLD_STORAGE_CONTENT_AVAILABLE_EVENT_NAME)) {
+            ColdStorageHelper.ColdStorageContentStatus coldStorageContentStatus = ColdStorageHelper.checkAvailabilityOfColdStorageContent(
+                    session);
+            assertEquals(session.getRepositoryName(), coldStorageContentStatus.getRepositoryName());
+
+            // all cold storage contents being retrieved are now available
+            assertEquals(documents.size(), coldStorageContentStatus.getTotalBeingRetrieved());
```

Comment:
If we want to keep `getTotalBeingRetrieved` (see https://github.com/nuxeo/nuxeo/pull/3666/commits/aa516a7728bc3b0ff6adf1b73cbe788845a41f8e#r374630558), I would expect the future retrieval process to set the "beingRetrieved" boolean to false, right?
Waiting for this, maybe we could decrement the beingRetrived counter in `checkAvailabilityOfColdStorageContent` along with `available++`, WDYT?
Just feels funny here to have as many documents available as the ones being retrieved.

Reply:
OK, so the boolean field is reset here: https://github.com/nuxeo/nuxeo/pull/3666/commits/7a25db5e1ed9270eebdd065e36a1ff12c5dce4f2#diff-ead64399e0b98c4a84e7b8cef49b5444R158.
So, I would definitley decrement the "beingRetrieved" counter just after.